### DISCUSSION TITLE
Relicensing certain files of libiio under the MIT license

### DISCUSSION THREAD

**Discussion by mhennerich** [analogdevicesinc/libiio#1210] [Announcements]
### Relicensing certain files of libiio under the MIT license

Dear libiio Contributors,
 
We hope this message finds you well.
 
As libiio continues to evolve and find new applications, we are considering relicensing certain files to better accommodate the diverse environments in which it is now being used.
As you may be aware, libiio is currently licensed under the LGPL-2.1 license, which has been well-suited for use within Linux environments. However, with the growing use of libiio in microcontroller environments where static linking is a requirement, and with plans to integrate with Zephyr, we have identified the need for a more permissive license on following files.

**[1] Files subject to relicense under the MIT license:**

- attr.c
- attr.h
- backend.c
- block.c
- buffer.c
- channel.c
- context.c
- device.c
- events.c
- iiod/ops.h
- iiod/responder.c
- iiod-responder.h
- iio-private.h
- include/iio/iio-backend.h
- include/iio/iio-debug.h
- include/iio/iio.h
- include/iio/iio-lock.h
- library.c
- lock-dummy.c
- mask.c
- scan.c
- sort.c
- sort.h
- stream.c
- task.c
- tinyiiod/tinyiiod.c
- tinyiiod/tinyiiod.h
- utilities.c
- xml.c

For the upcoming libiio v1.0 release, we have already introduced patches that enable support for compiling Libiio + IIOD on baremetal/no-OS embedded devices. Moreover, libiio can now be compiled without threads using the NO_THREADS CMake option, further extending its compatibility with various systems.

To this end, we are proposing to relicense those files [1] under the [MIT](https://spdx.org/licenses/MIT.html) license (SPDX short identifier: MIT). This change will allow for greater flexibility in how libiio/tinyiiod can be used, especially in scenarios where dynamic linking is not possible.

We believe relicensing these files [1] will not affect current users of libiio but also open new opportunities for its application in a wider range of projects.

To proceed with this change, we kindly request your written permission to relicense your contributions to libiio [1] under the MIT licenses.

**Your consent is essential for us to move forward with this initiative.**
Please let us know if you agree with this proposal by providing your written approval, or if you have any concerns or questions, feel free to reach out.

Thank you for your continued support and contributions to libiio.
We look forward to your response.
 
### What’s next?
We are in the process of reaching out to contributors of libiio [1] via email to discuss this important licensing update. However, we understand that sometimes emails may not reach their intended recipients due to various reasons.

That’s why you also find your github name mentioned here:

- @pcercuei
- @rgetz
- @gizero
- @abrauchli
- @catkira
- @spencercw
- @dpfrey
- @ekigwana
- @jtojnar
- @larsclausen
- @lucasrangit
- @mfornero
- @ynezz
- @rlefevre
- @rofferom
- @hshmt

We need the consent of the contributor, whether that’s you, or your employer at the time of the contribution. If we need to get consent from your employer at the time of contribution and that needs to come from someone else, please point us to the relevant point of contact so that we can follow up.
To express your agreement with relicensing your past contributions, we kindly ask you to either reply to the email sent to you, or to post the following statement in this thread by November 18th, 2024:
 
**“I irrevocably consent for my past contributions to libiio to be relicensed under the MIT license”**, indicating under which email address your contributions were made, if different from your current email address.
Your prompt response will greatly assist us in moving forward with this transition. If you do not agree with the proposed change, please respond with your comments as well. Open dialogue will help us address any concerns and proceed in a manner that respects the wishes of all contributors.
 
### What happens if I do not agree to relicense my code contributions?

Past contributions to the libiio project were made under the LGPL-2.1 license. If we do not get consent to relicense your past contributions, we will respect your copyright and rewrite or remove your contributions before relicensing these files [1]. It would be a disservice to your time and the project’s history to have to resort to rewriting your contributions, so if you have questions, please reply to the email or comment here.
 
Thank you for your attention to this matter and for your valuable contributions to libiio.
 
Best regards,
Michael Hennerich

**Comment by mhennerich** [comment]
I irrevocably consent for my past contributions to libiio to be relicensed under the MIT license.
Michael Hennerich

**Comment by lucasrangit** [comment]
I irrevocably consent for my past contributions to libiio to be relicensed
under the MIT license.

**Comment by spencercw** [comment]
I irrevocably consent for my past contributions to libiio to be relicensed under the MIT license.

**Comment by abrauchli** [comment]
I irrevocably consent for my past contributions to libiio to be relicensed under the MIT license

Andreas Brauchli, Nov 5, 2024

-------- Original Message --------

**Comment by rlefevre** [comment]
I irrevocably consent for my past contributions to libiio to be relicensed under the MIT license.
Rémi Lefèvre (remi.lefevre@parrot.com)

**Comment by dpfrey** [comment]
“I irrevocably consent for my past contributions to libiio to be relicensed
under the MIT license”

I either made my contributions as ***@***.*** or
***@***.***

Thanks for your hard work on libiio.

**Comment by ynezz** [comment]
I irrevocably consent for my past contributions to libiio to be relicensed under the MIT license.

**Comment by glen-anderson** [comment]
I irrevocably consent for my past contributions to libiio to be relicensed under the MIT license

**Comment by ekigwana** [comment]
I irrevocably consent for my past contributions to libiio to be re-licensed under the MIT license.
Edward Kigwana

**Comment by gizero** [comment]
I irrevocably consent for my past contributions to libiio to be relicensed under the MIT license

Andrea

**Comment by rofferom** [comment]
I irrevocably consent for my past contributions to libiio to be relicensed under the MIT license.

romain.roffe@parrot.com

**Comment by hshmt** [comment]
I irrevocably consent for my past contributions to libiio to be relicensed under the MIT license

**Comment by pcercuei** [comment]
I irrevocably consent for my past contributions to libiio to be relicensed under the MIT license

**Comment by larsclausen** [comment]
This is ok. I irrevocably consent for my past contributions to libiio to be relicensed under the MIT license

**Comment by catkira** [comment]
I irrevocably consent for my past contributions to libiio to be relicensed under the MIT license

**Comment by xR3b0rn** [comment]
I irrevocably consent for my past contributions to libiio to be relicensed under the MIT license.

**Comment by rgetz** [comment]
I irrevocably consent for my past personal contributions to libiio to be licensed under the MIT license as detailed at https://spdx.org/licenses/MIT.html  This does not include contributions while I was working at Analog Devices, nor MathWorks which needs to be re-licensed through those employers.

**Comment by mhennerich** [comment]
I am pleased to inform you that we have received consent from everyone regarding the **relicensing of certain files of libiio under the MIT license**, either through this thread or via direct email. Your prompt responses and cooperation are greatly appreciated.

Thank you all for your support and for making this happen. If there are any further updates or actions required, I will keep you informed.

Best regards,
Michael Hennerich
